### ISSUE TITLE
[PRE REVIEW]: Ripser.py: A Lean Persistent Homology Library for Python

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#901]
**Submitting author:** @sauln (<a href="http://orcid.org/0000-0002-8549-9810">Nathaniel Saul</a>)
**Repository:** <a href="https://github.com/scikit-tda/ripser.py" target ="_blank">https://github.com/scikit-tda/ripser.py</a>
**Version:** 0.2.4
**Editor:** @arokem
**Reviewers:** lmcinnes

**Author instructions**

Thanks for submitting your paper to JOSS @sauln. The JOSS editor (shown at the top of this issue) will work with you on this issue to find a reviewer for your submission before creating the main review issue.

@sauln if you have any suggestions for potential reviewers then please mention them here in this thread. In addition, [this list of people](https://bit.ly/joss-reviewers) have already agreed to review for JOSS and may be suitable for this submission.

**Editor instructions**

The JOSS submission bot @whedon is here to help you find and assign reviewers and start the main review. To find out what @whedon can do for you type:

```
@whedon commands
```

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @arokem it looks like you're currently assigned as the editor for this paper :tada:

For a list of things I can do to help you, just type:

```
@whedon commands
```

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
PDF failed to compile for issue #901 with the following error: 

 Can't find any papers to compile :-(

**Comment by arfon** [comment]
@sauln - please review the [submission guidelines](https://joss.readthedocs.io/en/latest/submitting.html) for JOSS. Your project doesn't seem to have a paper.

**Comment by sauln** [comment]
Hi @arfon, sorry the paper was hiding in the Joss branch. I’ve merged it into master now.

**Comment by sauln** [comment]
@whedon commands

**Comment by whedon** [comment]
Here are some things you can ask me to do:

```
# List Whedon's capabilities
@whedon commands

# List of editor GitHub usernames
@whedon list editors

# List of reviewers together with programming language preferences and domain expertise
@whedon list reviewers

🚧 🚧 🚧 Experimental Whedon features 🚧 🚧 🚧

# Compile the paper
@whedon generate pdf

```

**Comment by sauln** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
PDF failed to compile for issue #901 with the following error: 

 /app/vendor/ruby-2.3.4/lib/ruby/2.3.0/psych.rb:379:in `parse': (tmp/901/paper.md): did not find expected '-' indicator while parsing a block collection at line 8 column 2 (Psych::SyntaxError)
	from /app/vendor/ruby-2.3.4/lib/ruby/2.3.0/psych.rb:379:in `parse_stream'
	from /app/vendor/ruby-2.3.4/lib/ruby/2.3.0/psych.rb:327:in `parse'
	from /app/vendor/ruby-2.3.4/lib/ruby/2.3.0/psych.rb:254:in `load'
	from /app/vendor/ruby-2.3.4/lib/ruby/2.3.0/psych.rb:475:in `block in load_file'
	from /app/vendor/ruby-2.3.4/lib/ruby/2.3.0/psych.rb:474:in `open'
	from /app/vendor/ruby-2.3.4/lib/ruby/2.3.0/psych.rb:474:in `load_file'
	from /app/vendor/bundle/ruby/2.3.0/bundler/gems/whedon-e0f72c5e8125/lib/whedon.rb:68:in `initialize'
	from /app/vendor/bundle/ruby/2.3.0/bundler/gems/whedon-e0f72c5e8125/lib/whedon/processor.rb:32:in `new'
	from /app/vendor/bundle/ruby/2.3.0/bundler/gems/whedon-e0f72c5e8125/lib/whedon/processor.rb:32:in `set_paper'
	from /app/vendor/bundle/ruby/2.3.0/bundler/gems/whedon-e0f72c5e8125/bin/whedon:37:in `prepare'
	from /app/vendor/bundle/ruby/2.3.0/gems/thor-0.20.0/lib/thor/command.rb:27:in `run'
	from /app/vendor/bundle/ruby/2.3.0/gems/thor-0.20.0/lib/thor/invocation.rb:126:in `invoke_command'
	from /app/vendor/bundle/ruby/2.3.0/gems/thor-0.20.0/lib/thor.rb:387:in `dispatch'
	from /app/vendor/bundle/ruby/2.3.0/gems/thor-0.20.0/lib/thor/base.rb:466:in `start'
	from /app/vendor/bundle/ruby/2.3.0/bundler/gems/whedon-e0f72c5e8125/bin/whedon:99:in `<top (required)>'
	from /app/vendor/bundle/ruby/2.3.0/bin/whedon:22:in `load'
	from /app/vendor/bundle/ruby/2.3.0/bin/whedon:22:in `<main>'

**Comment by arfon** [comment]
@sauln - this PR should fix the paper build error: https://github.com/scikit-tda/ripser.py/pull/45

**Comment by sauln** [comment]
Thank you!  

@whedon generate pdf

**Comment by sauln** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[--> Check article proof :page_facing_up: <--](https://github.com/openjournals/joss-papers/blob/joss.00901/joss.00901/10.21105.joss.00901.pdf)

**Comment by sauln** [comment]
That is an impressive list of volunteers!

I think @Sealhuang, @sssomani, @abuccts, @thomasA, and @arokem all have interests and skills that would be relevant

**Comment by sauln** [comment]
I think perspectives from outside the field of TDA would be beneficial. We hope that the library will be accessible to anyone with just an interest in TDA, so understanding how someone without previous background approaches the library would be helpful for us. That being said, a JOSS review might not be the best forum for that.

**Comment by arokem** [comment]
@whedon assign arokem as editor

**Comment by whedon** [comment]
OK, the editor is arokem

**Comment by arokem** [comment]
👋 @lmcinnes: would you be willing to review this paper for JOSS?

**Comment by arokem** [comment]
👋 @maxrousseau: would you be willing to review this paper for JOSS?

**Comment by lmcinnes** [comment]
@arokem , I would be willing to review this, but may need a little assistance, having never reviewed for JOSS before.

**Comment by arokem** [comment]
Thanks @lmcinnes! There is documentation that explains the principles and process here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Let me know if you have questions. Don't be alarmed if it takes me a few days to get to your questions, though. For the time being, I will start the review, which will start another issue with a check-list for you to check the software against.

**Comment by arokem** [comment]
@whedon assign lmcinnes as reviewer

**Comment by whedon** [comment]
OK, the reviewer is lmcinnes

**Comment by arokem** [comment]
@whedon start review

**Comment by whedon** [comment]
OK, I've started the review over in https://github.com/openjournals/joss-reviews/issues/925. Feel free to close this issue now!

**Comment by arokem** [comment]
@lmcinnes : the check list and more guidance should now be available for you on #925. I'm closing this one.
